src/init.cc

```diff
@@ -1982,27 +1963,43 @@ static ncclResult_t ncclCommInitRankFunc(struct ncclAsyncJob* job_) {
       auto& mscclppUniqueId = mscclpp_uniqueIdMap[origUniqueId];
       mscclpp_uniqueIdMap[job->commId] = mscclppUniqueId;
       mscclpp_uniqueIdReverseMap[mscclppUniqueId].insert(job->commId);
+      ncclCommToUniqueIdMap[comm] = job->commId;
     }
   }
   else
 #endif
   if (rcclParamMscclppEnabled()) {
 #ifdef ENABLE_MSCCLPP
-    hipDeviceProp_t devProp;
-    CUDACHECK(hipGetDeviceProperties(&devProp, cudaDev));
-    comm->mscclppCompatible = IsArchMatch(devProp.gcnArchName, "gfx94");
-    if (comm->mscclppCompatible) {
-      auto& mscclppUniqueId = mscclpp_uniqueIdMap[job->commId];
-      NCCLCHECKGOTO(bootstrapIntraNodeBroadcast(comm->bootstrap, comm->localRankToRank, comm->localRank, comm->localRanks, 0, &mscclppUniqueId, sizeof(mscclppUniqueId)), res, fail);
-      unsigned long long mscclppUniqueIdHash; (void)mscclppUniqueIdHash;
-      TRACE_CALL("bootstrapIntraNodeBroadcast(rank=%d, nranks=%d, root=%d, bcastData=hash:0x%llx)", comm->localRank, comm->localRanks, 0, (mscclppUniqueIdHash = (unsigned long long)hashUniqueId(mscclppUniqueId)));
-      comm->mscclpp_threshold = rcclParamMscclppThreshold();
-      INFO(NCCL_INIT, "MSCCL++: Enabled! Msg size threshold=%zu", comm->mscclpp_threshold);
-      NCCLCHECKGOTO(mscclpp_ncclCommInitRank(&(comm->mscclpp_comm), job->nranks, mscclppUniqueId, job->myrank), res, fail);
-      TRACE_CALL("mscclpp_ncclCommInitRank (*comm=%p, nranks=%d, commId=hash:0x%llx, myrank=%d)", comm->mscclpp_comm, job->nranks, mscclppUniqueIdHash, job->myrank);
-      mscclpp_commToUniqueIdMap[comm->mscclpp_comm] = mscclppUniqueId;
+    if (mscclEnabled() && comm->topo->mscclEnabled && mscclppCommCompatible(comm)) {
```

Comment:
@nusislam Should it be this instead?

```suggestion
    if (mscclEnabled() && (comm->topo->mscclEnabled || mscclForceEnabled()) && mscclppCommCompatible(comm)) {
```